modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1810,10 +1845,19 @@ private CompletableFuture<Object> applyCmdWithExceptionHandling(Command cmd) {
      * Executes an Update command.
      *
      * @param cmd Update command.
-     * @return Raft future, see {@link #applyCmdWithExceptionHandling(Command)}.
+     * @return A local update ready future, possibly having a nested replication future as a result for delayed ack purpose.
      */
-    private CompletableFuture<Object> applyUpdateCommand(UpdateCommand cmd) {
+    private CompletableFuture<CompletableFuture<?>> applyUpdateCommand(UpdateCommand cmd) {
```

Comment:
I don't see it necessary to use this unclear logic. This method returns either a result or a replication future. A future wrapper is never required.

Reply:
Agree. Proposed API is overcomplicated.

Reply:
This signature is required to implement correct chaining both for implicit and explicit tx flows. 
The result of a call either the immediate result with a delayed replication future, or immediate replication future.
Actually this is the simplest way to implement it. 
Can you propose simpler variant, which is also correct ?